components/console-backend-service/internal/domain/assetstore/asset_converter.go

```diff
@@ -23,12 +27,17 @@ func (c *assetConverter) ToGQL(item *v1alpha2.Asset) (*gqlschema.Asset, error) {
 	}
 
 	status := c.extractor.Status(item.Status.CommonAssetStatus)
+	metadata, err := c.extractMetadata(item.Spec.Metadata)
```

Comment:
Please change name of type of `extractor` from `AssetStatusExtractor` to `AssetExtractor` and put there `extractMetadata` and use it here.